test/framework/src/main/java/org/elasticsearch/index/mapper/MapperServiceTestCase.java

```diff
@@ -827,30 +828,35 @@ protected final String syntheticSource(DocumentMapper mapper, CheckedConsumer<XC
      * That's the point, really. It'll just be "close enough" for
      * round tripping.
      */
-    private void roundTripSyntheticSource(DocumentMapper mapper, String syntheticSource, DirectoryReader reader) throws IOException {
+    private void roundTripSyntheticSource(MapperService mapperService, String syntheticSource, DirectoryReader reader) throws IOException {
         try (Directory roundTripDirectory = newDirectory()) {
             RandomIndexWriter roundTripIw = indexWriterForSyntheticSource(roundTripDirectory);
-            ParsedDocument doc = mapper.parse(new SourceToParse("1", new BytesArray(syntheticSource), XContentType.JSON));
+            ParsedDocument doc = mapperService.documentMapper()
+                .parse(new SourceToParse("1", new BytesArray(syntheticSource), XContentType.JSON));
             // Process root and nested documents in the same way as the normal indexing chain (assuming a single document)
             doc.updateSeqID(0, 0);
             doc.version().setLongValue(0);
             roundTripIw.addDocuments(doc.docs());
             roundTripIw.close();
             try (DirectoryReader roundTripReader = wrapInMockESDirectoryReader(DirectoryReader.open(roundTripDirectory))) {
-                String roundTripSyntheticSource = syntheticSource(mapper, roundTripReader, doc.docs().size() - 1);
+                String roundTripSyntheticSource = syntheticSource(mapperService, roundTripReader, doc.docs().size() - 1);
                 assertThat(roundTripSyntheticSource, equalTo(syntheticSource));
                 validateRoundTripReader(syntheticSource, reader, roundTripReader);
             }
         }
     }
 
-    protected static String syntheticSource(DocumentMapper mapper, IndexReader reader, int docId) throws IOException {
```

Comment:
I'd keep this flavor and pass empty `indexSettings` for the cases we don't care. That should reduce the churn for testing files..